HRC_TXING_CHECK/HRC_Txing_Check.py

```diff
@@ -171,67 +183,187 @@
+        event_hist_running = False
+        stop_science_date = each_cmd["date"]
+        stop_science_time = each_cmd["time"]
         
-        # Set the evh_found_flag to False so that you can find the next SI mode load.
-        evh_found_flag = False
+    # 215PCAOF - HRC Observation complete.
+    if ("215PCAOF" in  each_cmd["commands"]):
+        # Create the string that indicates the HRC obs 15V power down command issued.
+        HRC_shutdown_string = each_cmd["date"] + " 215PCAOF command: HRC Observation Ends"
         
-# Done scanning all the commands in the load. In a legal load, where each HRC Event Histogram SI mode
-# load is paired with an SCS-134 activation,   evh_found_flag == False at this point.
-# So if you get to this point and evh_found_flag == True, then there exists an Event Histogram load 
-# which has no corresponding scs-134 activation.  This check handles the case where there
-# was only one HRC science observation in the load but the SCS-134 activation is missing.
-if evh_found_flag == True:
-    # ERROR - One of the HRC Event Histograms were loaded but there was never a corresponding SCS-134 activation
-    full_comment = " ".join(("\n",">>> ERROR - HRC Observation Event Histogram SI mode was loaded at but SCS-134 was not activated.\n     SI Mode Load Start: ", bias_start_date))
-    print("\n",full_comment)
-
-    # Append the comment to the comment list
-    comment_list.append([bias_start_date, bias_start_time, full_comment])
+        # Now check to see if the event histogram was running at this point of
+        # the load. Several situations have to be covered:
+        #
+        #    1) The 215PCAOF comes before the EVHIST stop science (i.e. EVHIST
+        #        still running) - always OK.
+        #
+        #    2) SCS-134 was never activated - ERROR because if you are stopping HRC you
+        #        must have expected that it was started
+        #
+        #    3) The EVHIST stop science comes before the 215PCAOF command
+        #         -  Delta t between the  AA00 and 215PCAOF commands <= 1 second - OK
+        #         -  Delta t between the  AA00 and 215PCAOF commands > 1 second - NOT OK
+        #                - See Guideline
+        #
+        #
+        #
+        # Possibility #1 - EVHIST still running
+        if event_hist_running == True:
+            comment_list.append([each_cmd["date"], each_cmd["time"], HRC_shutdown_string])
+            
+        # Possibility #2 - Check to see if the SCS-134 activation ever occurred
+        elif hrc_activated_flag == False:
+            HRC_activation_error_string = " ".join((HRC_shutdown_string, "\n>>> ERROR - SCS-134  was never activated"))
+            print("\n", HRC_activation_error_string)
+            comment_list.append([each_cmd["date"], each_cmd["time"], HRC_activation_error_string])
+
+        # Possibility #3  - EVHIST loaded, run, and was shut down prior to 215PCAOF
+        elif (evh_loaded_flag == True) and \
```

Comment:
There is a hole in the logic gates here for hrc_activated_flag == True and evh_loaded_flag == False.  This should have kicked up errors earlier, but do you want to do anything with such an instance here also in your elifs?

Reply:
This contingency is colvered by a previous check in the code.